cloudfoundry-client-reactor/src/test/java/org/cloudfoundry/reactor/uaa/clients/ReactorClientsTest.java

```diff
@@ -53,6 +56,76 @@
 
 public final class ReactorClientsTest {
 
+    public static final class BatchDelete extends AbstractUaaApiTest<BatchDeleteClientsRequest, BatchDeleteClientsResponse> {
+
+        private final ReactorClients clients = new ReactorClients(CONNECTION_CONTEXT, this.root, TOKEN_PROVIDER);
+
+        @Override
+        protected InteractionContext getInteractionContext() {
+            return InteractionContext.builder()
+                .request(TestRequest.builder()
+                    .method(POST).path("/oauth/clients/tx/delete")
+                    .payload("fixtures/uaa/clients/POST_tx_delete_request.json")
+                    .build())
+                .response(TestResponse.builder()
+                    .status(OK)
+                    .payload("fixtures/uaa/clients/POST_tx_delete_response.json")
+                    .build())
+                .build();
+        }
+
+        @Override
+        protected BatchDeleteClientsResponse getResponse() {
+            return BatchDeleteClientsResponse.builder()
+                .client(Client.builder()
+                    .approvalsDeleted(true)
+                    .allowedProvider("uaa", "ldap", "my-saml-provider")
+                    .authority("clients.read", "clients.write")
+                    .authorizedGrantType(CLIENT_CREDENTIALS)
+                    .autoApprove("true")
+                    .clientId("14pnUs")
+                    .lastModified(1468364444461L)
+                    .name("My Client Name")
+                    .redirectUriPattern("http*://ant.path.wildcard/**/passback/*", "http://test1.com")
+                    .resourceId("none")
+                    .scope("clients.read", "clients.write")
+                    .tokenSalt("erRsWH")
+                    .build())
+                .client(Client.builder()
+                    .approvalsDeleted(true)
+                    .allowedProvider("uaa", "ldap", "my-saml-provider")
+                    .authority("clients.read", "clients.write")
+                    .authorizedGrantType(CLIENT_CREDENTIALS)
+                    .autoApprove("true")
+                    .clientId("qECLyr")
+                    .lastModified(1468364444868L)
+                    .name("My Client Name")
+                    .redirectUriPattern("http*://ant.path.wildcard/**/passback/*", "http://test1.com")
+                    .resourceId("none")
+                    .scope("clients.read", "clients.write")
+                    .tokenSalt("48TIsq")
+                    .build())
+                .build();
+        }
+
+        @Override
+        protected BatchDeleteClientsRequest getValidRequest() throws Exception {
+            return BatchDeleteClientsRequest.builder()
+                .clientIdentification(ClientIdentification.builder()
```

Comment:
I'm not sure I like this interface—it is wordy and a little cumbersome for the user.  Also there is no prospect of the `ClientIdentification` class being used anywhere else.  I'll work on a way of simplifying this.